packages/discord.js/src/structures/ButtonBuilder.js

```diff
@@ -5,8 +5,10 @@ const Transformers = require('../util/Transformers');
 const Util = require('../util/Util');
 
 class ButtonBuilder extends BuildersButtonComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+  constructor({ emoji, ...data }) {
+    super(
+      Transformers.toSnakeCase({ ...data, emoji: emoji && typeof emoji === 'string' ? Util.parseEmoji(emoji) : emoji }),
```

Comment:
Is the `emoji &&` check intentionally there to exclude empty strings (`''`) from being parsed? Are they valid?

Reply:
I believe this was actually done to check for truthy values as I may have added the typeof check later on and forgot to remove that part. Should I remove it then or is it fine to keep?

Reply:
If `''` is somehow valid, then this code is automatically valid, otherwise remove.